### ISSUE TITLE
Firefox crashes on OnShape compatability check when using primus on Fedora 25

### ISSUE THREAD

**Issue by gregoryleblanc** [Bumblebee-Project/Bumblebee#858]
I can consistently crash firefox by viewing the OnShape compatibility check page https://cad.onshape.com/check using Firefox when launched with primusrun on Fedora 25.
optirun firefox seems to work just fine so far (with tremendous reported performance increases)

Details:
Fedora 25
$ uname -a
Linux localhost.localdomain 4.9.14-200.fc25.x86_64 #1 SMP Mon Mar 13 19:26:40 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

$ rpm -q fedora-release
fedora-release-25-1.noarch

Manufacturer and model:

baseboard-manufacturer: Dell Inc.
baseboard-product-name: 0T3YTY
baseboard-version     : A00
system-manufacturer   : Dell Inc.
system-product-name   : Precision M4800
system-version        : 01
bios-vendor           : Dell Inc.
bios-version          : A13
bios-release-date     : 01/29/2015

$ optirun -V
optirun (Bumblebee) 3.2.1

[bumblebee-bugreport-20170328_200640.tar.gz](https://github.com/Bumblebee-Project/Bumblebee/files/877394/bumblebee-bugreport-20170328_200640.tar.gz)

**Comment by gregoryleblanc** [comment]
$ primusrun firefox
console.warn: SDK worker-child started as frozen on unexpected initial document.readyState {"initialDocumentReadyState":"uninitialized","windowLocation":"about:blank"}
Gdk-Message: firefox: Fatal IO error 11 (Resource temporarily unavailable) on X server :8.

**Comment by gsgatlin** [comment]
Hello. Normally I would suggest opening a issue on the primus issue tracker. But I am unsure if the developer is still active. I think only he really understands the code.

@amonakov Do you have any ideas on how @gregoryleblanc or I can troubleshoot the issue further?

Just so I understand, you are trying to play some 3d games in a web browser like firefox?

**Comment by gregoryleblanc** [comment]
No, I'm trying to use a 3D modeling software that runs in the browser using WebGL.  

I didn't even realize that primus was a separate thing from bumblebee.  All my my knowledge comes from reading online for the past week, starting with the Fedora Bumblebee wiki page.

**Comment by gsgatlin** [comment]
@gregoryleblanc Oh thats ok. His software is at https://github.com/amonakov/primus/ and "primusrun" is part of that software.
